api/controllers/game/spectate/join.js

```diff
@@ -0,0 +1,56 @@
+module.exports = async function (req, res) {
+  let { gameId } = req.params;
+  gameId = Number(gameId);
+
+  try {
+    const [ spectator, game ] = await Promise.all([
+      User.findOne({ id: req.session.usr }),
+      Game.findOne(gameId).populate('gameStates')
+        .populate('p0')
+        .populate('p1'),
```

Comment:
Originally I had this using lockGame, but I don't think it's necessary for spectators to lock the game by joining (or leaving) so I replaced it with a raw .findOne()